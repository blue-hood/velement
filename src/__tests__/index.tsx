import '@testing-library/jest-dom/extend-expect';

import VirtualElement, { appendChildren, createElement } from '..';

interface DivProps {
  style?: string;
}

class Div extends VirtualElement<HTMLDivElement> {
  public constructor(element: HTMLDivElement | null, props: DivProps) {
    super(element || 'div');

    if (props.style) {
      this.element.setAttribute('style', props.style);
    }

    this.element.innerHTML = 'VirtualDivElement';
  }
}

declare module '..' {
  function createElement(type: typeof Div, props: DivProps, ...children: Child[]): Div;
}

test('should append children', async () => {
  const div = document.createElement('div');

  const htmlElement = createElement('div', null);
  htmlElement.innerHTML = `HTMLDivElement`;

  const virtualElement = createElement(Div, {});

  appendChildren(div, 'TextNode', htmlElement, virtualElement);

  expect(div).toMatchSnapshot();
});

test('should create HTMLElement', async () => {
  const div = createElement('div', {
    style: 'font-weight: bold;'
  });
  div.innerHTML = `HTMLDivElement`;

  expect(div).toMatchSnapshot();
});

test('should create VirtualElement', async () => {
  const div = createElement(Div, {
    style: 'font-weight: bold;'
  });

  expect(div.element).toMatchSnapshot();
});

test('should compose HTMLElement', async () => {
  const htmlElement = document.createElement('div');
  const virtualElement = new Div(htmlElement, {});

  expect(virtualElement.element).toMatchSnapshot();
});
