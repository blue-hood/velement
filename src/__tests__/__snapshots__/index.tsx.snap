// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should append children 1`] = `
<div>
  TextNode
  <div>
    HTMLDivElement
  </div>
  <div>
    VirtualDivElement
  </div>
</div>
`;

exports[`should compose HTMLElement 1`] = `
<div>
  VirtualDivElement
</div>
`;

exports[`should create HTMLElement 1`] = `
<div
  style="font-weight: bold;"
>
  HTMLDivElement
</div>
`;

exports[`should create VirtualElement 1`] = `
<div
  style="font-weight: bold;"
>
  VirtualDivElement
</div>
`;
